shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -49,20 +57,41 @@ public PlatformViewsController() {
         vdControllers = new HashMap<>();
     }
 
-    public void attachFlutterView(FlutterView view) {
-        if (mFlutterView != null)
+    /**
+     * Attaches this platform views controller to its input and output channels.
+     *
+     * @param context The base context that will be passed to embedded views created by this controller.
+     *                This should be the context of the Fragment or Activity hosting the Flutter applicaiton.
+     * @param textureRegistry The texture registry which provides the output textures into which the embedded views
+     *                        will be rendered.
+     * @param messenger The Flutter application on the other side of this messenger drives this platform views controller.
+     */
+    public void attach(Context context, TextureRegistry textureRegistry, BinaryMessenger messenger) {
```

Comment:
We spoke offline about this, but I think it would be helpful to ensure that all "attach" methods we declare throughout the embedding somehow specifies what they attach to.  This will avoid ambiguity.  For example: "onAttachedToWindow()", "attachToFlutterRenderer()", etc.

I'm not sure what the right name for this one would be.  Maybe this is also attaching to a Flutter renderer?  Maybe this attaching to a Flutter engine?  I'm not sure what the overriding conceptual artifact is, but whatever it is would be helpful.

This is a non-blocking concern.  Feel free to merge and improve later, if needed.

Reply:
Open for concrete suggestions of a better name.